examples/application/main.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"../../iis"
+)
+
+func main() {
+	log.Printf("Starting test...")
+
+	testAppPath := "C:\\inetpub\\testing"
+	testAppContent := "<html><head><title>IIS Testing</title></head><body><h1>Hello World!</h1></body></html>"
+	appPoolName := "testing"
+	applicationName := "testing"
+	websiteName := "Default Web Site"
+	testUrl := "http://localhost/testing"
+
+	err := createTestDirectory(testAppPath, testAppContent)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("Creating IIS Client")
+	client, err := iis.NewClient()
+	if err != nil {
+		log.Fatalf("Error creating IIS client: %s", err)
+	}
+
+	log.Printf("Creating IIS App Pool %q", appPoolName)
+	err = createAppPool(client, appPoolName)
+	if err != nil {
+		_ = os.RemoveAll(testAppPath)
+		log.Fatal(err)
+	}
+
+	log.Printf("Adding IIS Application %s to the Default Web Site", "testing")
+	err = createApplication(client, applicationName, appPoolName, testAppPath, websiteName)
+	if err != nil {
+		_ = client.AppPools.Delete(appPoolName)
+		_ = os.RemoveAll(testAppPath)
+		log.Fatal(err)
+	}
+
+	log.Printf("Trying to get the test HTML from IIS %s", testUrl)
+	testApplication(testUrl, testAppContent)
+
+	log.Printf("Cleaing up...")
+
+	cleanup(client, applicationName, websiteName, appPoolName, testAppPath)
+
+	log.Printf("Done!")
+}
+
+func cleanup(client *iis.Client, applicationName string, websiteName string, appPoolName string, testAppPath string) {
+	err := client.Applications.Delete(applicationName, websiteName)
+	if err != nil {
+		log.Printf("Error removing Application: %s", err)
+		err = nil
+	}
+	err = client.AppPools.Delete(appPoolName)
+	if err != nil {
+		log.Printf("Error removing App Pool: %s", err)
+		err = nil
```

Comment:
and then return the error here:

```suggestion
		return log.Errorf("Error removing App Pool: %s", err)
```